### PR TITLE
fix: routing for new-style partner urls

### PR DESCRIPTION
**Pull request by anandaroop**
The type of this PR is: **Bugfix**

This PR resolves [CX-1932] and [FX-3279]

### Description

- New-style partner pages were [launched a while ago ](https://github.com/artsy/force/pull/7682)
- The url structure for these changed as well (to have more conventional URLs rather than relying on Force's old vanity profile url pattern)
- The routing logic in Eigen was not updated accordingly

This PR adds the missing routing logic and 
- handles all the currently supported partner URLs as inferred from [Force's own redirection logic](https://github.com/artsy/force/blob/master/src/desktop/apps/partner_redirect/partnerRedirectionMiddleware.ts)
- sends em all to the native Partner screen (there are no separate entry points / query renderers for partner artworks or partner artists, so the Partner screen will have to suffice for now I believe)

#### Here is the reported issue, fixed now:

https://user-images.githubusercontent.com/140521/133161434-e4c437bd-523c-4629-a720-62ff78c2715f.mov

#### Here are the three permutations of partner urls, fixed now:

https://user-images.githubusercontent.com/140521/133161433-74c89795-99ba-4241-917b-0d07db1a7f55.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix deep link routing for new-style partner URLs - Roop @anandaroop

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1932]: https://artsyproduct.atlassian.net/browse/CX-1932
[FX-3279]: https://artsyproduct.atlassian.net/browse/FX-3279

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 17fdcc31026f6f306c512bb23eb109b217987693
</p>

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -113,6 +84,7 @@
                 <data android:pathPrefix="/my-profile" />
                 <data android:pathPrefix="/my-profile/saved-addresses" />
                 <data android:pathPrefix="/orders" />
+                <data android:pathPrefix="/partner" />
```
Comment (anandaroop) [inline android/app/src/main/AndroidManifest.xml:87]: Hmm lot of formatting noise when I updated this file, but this^ is the important line

**Review by gkartalis — APPROVED** [review]
🚀

**Comment by anandaroop** [comment]
@brainbicycle I'll go ahead and merge this too so that it's in the next release

**Pull request merged**